collins/collins.go

```diff
@@ -99,14 +100,20 @@ func (e *Error) Error() string {
 
 // NewClient creates a Client struct and returns a point to it. This client is
 // then used to query the various APIs collins provides.
-func NewClient(username, password, baseurl string) (*Client, error) {
+func NewClient(username, password, baseurl string, timeout int) (*Client, error) {
```

Comment:
This is a breaking change though because now everybody needs to provide this timeout parameter. Also, if the default until now was timeout of `0`, I think you should respect such a value and only default to 60 if negative number is given?

Reply:
the dangers of a v0 library.

An alternate style that would preserve earlier signatures is to implement timeout as a variadic option like in a [few](https://godoc.org/go.uber.org/dig#New) [other](https://godoc.org/go.uber.org/zap#New) [libraries](https://godoc.org/github.com/uber/jaeger-client-go#NewRemotelyControlledSampler) that you _may_ be familiar with.

Reply:
Good idea, turning this into a variadic would be much more flexible going forward as well if we wanted to let people overwrite the entire http client as well in the future.